pkg/volume/util/resize_util.go

```diff
@@ -46,13 +48,85 @@ func ClaimToClaimKey(claim *v1.PersistentVolumeClaim) string {
 	return fmt.Sprintf("%s/%s", claim.Namespace, claim.Name)
 }
 
+// UpdatePVSize updates just pv size after cloudprovider resizing is successful
+func UpdatePVSize(
+	pv *v1.PersistentVolume,
+	newSize resource.Quantity,
+	kubeClient clientset.Interface) error {
+	pvClone := pv.DeepCopy()
+
+	oldData, err := json.Marshal(pvClone)
+	if err != nil {
+		return fmt.Errorf("Unexpected error marshaling old PV %q with error : %v", pvClone.Name, err)
+	}
+
+	pvClone.Spec.Capacity[v1.ResourceStorage] = newSize
+
+	newData, err := json.Marshal(pvClone)
+	if err != nil {
+		return fmt.Errorf("Unexpected error marshaling new PV %q with error : %v", pvClone.Name, err)
+	}
+
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, pvClone)
+	if err != nil {
+		return fmt.Errorf("Error Creating two way merge patch for PV %q with error : %v", pvClone.Name, err)
+	}
+
+	_, err = kubeClient.CoreV1().PersistentVolumes().Patch(pvClone.Name, types.StrategicMergePatchType, patchBytes)
+	if err != nil {
+		return fmt.Errorf("Error Patching PV %q with error : %v", pvClone.Name, err)
+	}
+	return nil
+}
+
+// MarkResizeInProgress marks cloudprovider resizing as in progress
+func MarkResizeInProgress(
+	pvc *v1.PersistentVolumeClaim,
+	kubeClient clientset.Interface) (*v1.PersistentVolumeClaim, error) {
+	// Mark PVC as Resize Started
+	progressCondition := v1.PersistentVolumeClaimCondition{
+		Type:               v1.PersistentVolumeClaimResizing,
+		Status:             v1.ConditionTrue,
+		LastTransitionTime: metav1.Now(),
+	}
+	conditions := []v1.PersistentVolumeClaimCondition{progressCondition}
+	newPVC := pvc.DeepCopy()
+	newPVC = MergeResizeConditionOnPVC(newPVC, conditions)
+	return PatchPVCStatus(pvc /*oldPVC*/, newPVC, kubeClient)
+}
+
+// MarkForFSResize marks file system resizing as pending
+func MarkForFSResize(
+	pvc *v1.PersistentVolumeClaim,
+	kubeClient clientset.Interface) error {
+	pvcCondition := v1.PersistentVolumeClaimCondition{
+		Type:               v1.PersistentVolumeClaimFileSystemResizePending,
+		Status:             v1.ConditionTrue,
+		LastTransitionTime: metav1.Now(),
+		Message:            "Waiting for user to (re-)start a pod to finish file system resize of volume on node.",
+	}
+	conditions := []v1.PersistentVolumeClaimCondition{pvcCondition}
+	newPVC := pvc.DeepCopy()
+	newPVC = MergeResizeConditionOnPVC(newPVC, conditions)
+	_, err := PatchPVCStatus(pvc /*oldPVC*/, newPVC, kubeClient)
+	return err
+}
```

Comment:
We need to add version check to PVC patches. Same as - https://github.com/kubernetes/kubernetes/pull/71611/files#diff-a2fc41851c6b643da9fac7f382cec784R64 does.

Reply:
let's just use Update/UpdateStatus everywhere

Reply:
I am not sure about that. @liggitt warned us about safety net against version skews.

Reply:
gonna need some explanation/link of/to that issue (:... The controller gets a pvc from the informer then `Update`s it, where does version skew come into it?

Reply:
an update by an old client would drop a new field. patch of specific fields would preserve unknown fields.

Reply:
ok, understood, it's for kubelet's sake that we have to use `Patch` instead of `Update` since kubelet uses some of these functions & kubelet could be older than master. Will include resourceVersion in every `Patch`.